packages/venia-ui/lib/components/Image/image.js

```diff
@@ -75,18 +113,48 @@ const Image = props => {
     );
 };
 
+const conditionallyRequiredString = (props, propName, componentName) => {
+    // This component needs one of src or resource to be provided.
+    if (!props.src && !props.resource) {
+        return new Error(
+            `Missing both 'src' and 'resource' props in ${componentName}. ${componentName} needs at least one of these to be provided.`
+        );
+    }
+
+    return PropTypes.checkPropTypes(
+        {
+            resource: string,
+            src: string
+        },
+        props,
+        propName,
+        componentName
+    );
+};
+
 Image.propTypes = {
     alt: string,
     classes: shape({
         loaded: string,
         notLoaded: string,
         root: string
     }),
+    height: oneOfType([number, string]),
     onError: func,
     onLoad: func,
     placeholder: string,
-    src: string,
-    fileSrc: string
+    resource: conditionallyRequiredString,
+    resourceHeight: oneOfType([number, string]),
+    resourceWidth: oneOfType([number, string]),
```

Comment:
Sadly not, I had to separate them.

`resourceWidth` and `resourceHeight` are used for the fallback `src` for browsers that don't support `srcset` / `sizes`.

```jsx
<Image resource="resource.jpg" resourceWidth="80" resourceHeight="100" />
```

will result in something like `<img src="resource.jpg?width=80&height=100">`.

But that `img` can be styled differently. For example, it may have `width: 45px`. In English, "request the width 80 image, and put it into this 45px space".

Often the CSS widths are dependent on the viewport / parent elements, so these two values don't always match up.